### PR TITLE
Update faas-memory to use go modules

### PR DESCRIPTION
**Pull request by nmhancock**
This commit updates faas-memory to build with go modules
It also updates handlers/info.go to work with the latest provider types.
Finally it updates main.go for greater symmetry with faas-netes'.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module github.com/openfaas-incubator/faas-memory
+
+go 1.19
```
Comment (alexellis) [inline go.mod]: This should be go 1.18 to match the rest of OpenFaaS. We will get to 1.19, but need more time.

Whatever value is set here is a minimum version for any code that imports provided packages.

Comment (nmhancock) [reply go.mod]: fixed :)

**Comment by alexellis** [comment]
Would you like to add CI as per alexellis/arkade so that we have a build definition to validate PRs?

**Comment by alexellis** [comment]
The Dockerfile could also be improved as per openfaas/faas-netes so that it's multi-arch compatible

**Comment by nmhancock** [comment]
That and the README too, I'll see what I can do :)

**Comment by alexellis** [comment]
We also need an issue template and PR template from openfaas/faas

**Pull request merged**